sykle/call_subprocess.py

```diff
@@ -84,4 +88,32 @@ def __exit__(self, *args):
         pass
 
 
+class SubprocessExceptionHandler:
+    def __init__(self):
+        self.exc_stack = []
+
+    def push(self, exc):
+        self.exc_stack.append(exc)
+
+    def raise_exc(self):
+        # Gather the errors' stack traces and return
+        # them in a single final error.
+        if len(self.exc_stack):
+            stacktraces = ''
+            for idx, error in enumerate(self.exc_stack):
+                # Acrobatics to get traceback to capture
+                # and format each error's stack trace.
+                try:
+                    raise error
+                except error.__class__:
+                    stacktraces += "%s%s" % (
+                        traceback.format_exc(),
```

Comment:
Is there no way to get the stack trace from the error itself? I’m assuming not, since you did this, but I think this will add this line to the stacktrace, right?

Reply:
made it better